core/os/rw_lock.h

```diff
@@ -59,6 +59,8 @@ class RWLockRead {
 	RWLockRead(const RWLock *p_lock) {
 		if (p_lock) {
 			lock = const_cast<RWLock *>(p_lock);
+		} else {
+			lock = 0;
```

Comment:
Is this bad for performance @hpvb . Seems like having lock uninitialized is risky.

Reply:
Would `const_cast<RWLock *>(p_lock)` fail if `p_lock` is `NULL`? Otherwise we could just forego the `if (p_lock)` as done in `RWLockWrite`.

Reply:
Yeah, we should definitely do that.
const_cast with NULL args will return a NULL pointer to `RWLock`, so yeah, skip the if.